### ISSUE TITLE
lowering / cost analysis of @nnx.jit functions

### ISSUE THREAD

**Issue by cgarciae** [google/flax#4094]
### Discussed in https://github.com/google/flax/discussions/4093

<div type='discussions-op-text'>

<sup>Originally posted by **lzanini** July 19, 2024</sup>
hi, is there a way to use jax's cost analysis api for nnx.jitted functions ? typically, a training step with a module, an optimizer and a metric. it is unclear to me how to lower the inner jax.jitted function.</div>

**Comment by cgarciae** [comment]
Currently the inner jitted function is store in an `.inner` attribute.

```python
f = nnx.jit(lambda x: x)

print(f.inner). # <PjitFunction of <function jit_fn at ...>>
```
However, there is no easy way to lower with the exact same inputs that NNX will pass to it. We should provide some way to do this.

**Comment by cgarciae** [comment]
@gabbard sure! Happy to review a PR.

**Comment by jkyl** [comment]
Hi, I was just wondering if this PR ever landed?

**Comment by qGentry** [comment]
Hi guys, I'm also wondering how can I approach it, any news? 

@cgarciae @gabbard

**Comment by qGentry** [comment]
I've just upgraded to latest flax (0.12.0) and seems like now nnx.jitted function now has .lower exposed so I guess this issue can be marked as resolved

**Comment by vfdev-5** [comment]
We can also use pure jax transforms with nnx.Modules since 0.11.
Closing as resolved.
